### PR TITLE
Add SegmentedControl support

### PR DESCRIPTION
**Pull request by cristianames92**
This PR intents to add UISegmentedControl to the framework, wrapped in Segmented component

### CODE REVIEW

File: Sources/Components/Segmented.swift [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+//
+//  Created by Cristian Ames on 4/4/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+import UIKit
+
+public struct SegmentedProperties<MessageType> {
+    
+    public var segments: [SegmentProperties<MessageType>]
```
Comment (guidomb) [inline Sources/Components/Segmented.swift]: Lets use the proposed alternative of using a zip list.

```swift
public var leftSegments: [SegmentProperties<MessageType>]
public var selectedSegment: SegmentProperties<MessageType>
public var rightSegments: [SegmentProperties<MessageType>]
```

File: Sources/UIKit/Renderers/ButtonRenderer.swift [hunk 2]
```diff
@@ -41,7 +41,7 @@ internal struct ButtonRenderer<MessageType>: UIKitRenderer {
 extension UIButton {
     
     fileprivate func dispatch<MessageType>(message: MessageType, for event: UIControlEvents, with mailbox: Mailbox<MessageType> = Mailbox()) -> Mailbox<MessageType> {
-        let dispatcher = MessageDispatcher(message: message, mailbox: mailbox)
+        let dispatcher = MessageDispatcher(mailbox: mailbox) { _ in return message }
```
Comment (guidomb) [inline Sources/UIKit/Renderers/ButtonRenderer.swift]: Here we could use the `init(mailbox:,message:)` initializer

File: Sources/UIKit/Renderers/SegmentedRendered.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+//  PortalView
+//
+//  Created by Cristian Ames on 4/4/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+import UIKit
+import Foundation
+
+//public let defaultButtonFontSize = UInt(UIFont.buttonFontSize)
```
Comment (guidomb) [inline Sources/UIKit/Renderers/SegmentedRendered.swift]: Delete this

File: Sources/UIKit/Renderers/SegmentedRendered.swift [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                for: .valueChanged, with: mailbox
+            )
+        }
+        
+        return Render(view: segmentedControl, mailbox: mailbox)
+    }
+    
+}
+
+internal struct SegmentedControlDispatcher {
```
Comment (guidomb) [inline Sources/UIKit/Renderers/SegmentedRendered.swift]: remove this

File: Sources/UIKit/Renderers/SegmentedRendered.swift [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    
+    fileprivate func dispatch<MessageType>(messages: [MessageType?], for event: UIControlEvents, with mailbox: Mailbox<MessageType> = Mailbox()) -> Mailbox<MessageType> {
+        
+        let dispatcher = MessageDispatcher(mailbox: mailbox) { sender in
+            guard let segmentedControl = sender as? UISegmentedControl else { return .none }
+            let index = segmentedControl.selectedSegmentIndex
+            return index < messages.count ? messages[index] : .none
+        }
+        self.register(dispatcher: dispatcher)
+        self.addTarget(dispatcher, action: #selector(MessageDispatcher<MessageType>.dispatch), for: event)
```
Comment (guidomb) [inline Sources/UIKit/Renderers/SegmentedRendered.swift]: just in case I would remove all previous registered targets.

**Review by guidomb — CHANGES_REQUESTED** [review]

File: Sources/UIKit/Renderers/SegmentedRendered.swift (hunk 5 above)
Comment (cristianames92) [reply Sources/UIKit/Renderers/SegmentedRendered.swift]: Targets are being removed here https://github.com/cristianames92/PortalView/blob/master/Sources/UIKit/Renderers/SegmentedRenderer.swift#L40

**Review by guidomb — APPROVED** [review]

**Pull request merged**